autogpt_platform/backend/backend/blocks/slantthreed/webhook.py

```diff
@@ -0,0 +1,125 @@
+from pydantic import BaseModel
+
+from backend.data.block import (
+    Block,
+    BlockCategory,
+    BlockOutput,
+    BlockSchema,
+    BlockWebhookConfig,
+)
+from backend.data.model import SchemaField
+from backend.util import settings
+from backend.util.settings import BehaveAs
+
+from ._api import (
+    TEST_CREDENTIALS,
+    TEST_CREDENTIALS_INPUT,
+    Slant3DCredentialsField,
+    Slant3DCredentialsInput,
+)
+
+
+class Slant3DTriggerBase:
+    """Base class for Slant3D webhook triggers"""
+
+    class Input(BlockSchema):
+        credentials: Slant3DCredentialsInput = Slant3DCredentialsField()
+        # Webhook URL is handled by the webhook system
+        payload: dict = SchemaField(hidden=True, default={})
+
+    class Output(BlockSchema):
+        payload: dict = SchemaField(
+            description="The complete webhook payload received from Slant3D"
+        )
+        order_id: str = SchemaField(description="The ID of the affected order")
+        error: str = SchemaField(
+            description="Error message if payload processing failed"
+        )
+
+    def run(self, input_data: Input, **kwargs) -> BlockOutput:
+        yield "payload", input_data.payload
+        yield "order_id", input_data.payload["orderId"]
+
+
+class Slant3DOrderWebhookBlock(Slant3DTriggerBase, Block):
+    """Block for handling Slant3D order webhooks"""
+
+    class Input(Slant3DTriggerBase.Input):
+        class EventsFilter(BaseModel):
+            """
+            Currently Slant3D only supports 'SHIPPED' status updates
+            Could be expanded in the future with more status types
+            """
+
+            shipped: bool = True
+
+        events: EventsFilter = SchemaField(
+            title="Events", description="Order status events to subscribe to"
+        )
```

Comment:
You can give this filter a default value and hide it, so it doesn't show up on the block. I'll make a ticket to make the event filter optional as a structural solution.